### DISCUSSION TITLE
Proposal: otherwise

### DISCUSSION THREAD

**Discussion by VoteForPedro** [dotnet/csharplang#8171] [Language Ideas]
General Idea
The otherwise keyword is introduced to complement the existing if-else syntax. It provides a more natural language approach to conditional statements, making the code easier to read and understand. The otherwise keyword is designed to be used in conjunction with if and else if statements to handle alternative conditions.

```
if (condition1)
{
    // Code to execute if condition1 is true
}
otherwise
{
    // Code to execute if condition1 is false
}
else if (condition2)
{
    // Code to execute if condition1 is false and condition2 is true
}
otherwise
{
    // Code to execute if condition2 is false
}
else
{
    // Code to execute if none of the above conditions are true
}
```

Semantics
if (condition): Executes the block of code if the condition is true.
otherwise: Executes the block of code if the preceding if or else if condition is false. This provides an alternative action specific to the previous condition.
else if (condition): Executes the block of code if the initial if condition is false and this new condition is true.
else: Executes the block of code if all preceding if and else if conditions are false.

Benefits
Improved Readability
Clearer Intent: It clarifies the programmer's intent by explicitly separating different alternative actions associated with specific conditions.
Maintainability: Easier to maintain and understand complex conditional structures

Potential Issues
The existing else keyword would need to coexist with otherwise, which could potentially confuse some developers

**Comment by huoyaoyuan** [comment]
> Improved Readability
> Clearer Intent

I can't understand the execute conditions of each branch by reading your code and comment.
It seems to transform to:

```
if (condition1)
{
    // Code to execute if condition1 is true
}
else
{
    // Code to execute if condition1 is false
    if (condition2)
    {
        // Code to execute if condition1 is false and condition2 is true
    }
    else
    {
        // Code to execute if condition2 is false
        if (condition3)
        {
            // Code to execute if condition1 is false and condition2 is false and condition3 is true
        }
        else
        {
            // Code to execute if condition3 is false
            // Code to execute if none of the above conditions are true
        }
    }
}
```

Or if the block is extracted as a function:

```
if (condition1)
{
    // Code to execute if condition1 is true
    return;
}
// Code to execute if condition1 is false
if (condition2)
{
    // Code to execute if condition1 is false and condition2 is true
    return;
}
// Code to execute if condition2 is false
if (condition3)
{
    // Code to execute if condition1 is false and condition2 is false and condition3 is true
    return;
}
// Code to execute if condition3 is false
// Code to execute if none of the above conditions are true
```

However, it is not so common to have such statements to execute before testing for next condition. If there is any, the existing forms are more understandable.

**Comment by VoteForPedro** [comment]
Thanks for the feedback. The pointlessness of this idea has definitely become clear to me.
